File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0;
        int r = heights.size() - 1;

        int max = 0;
        while (l < r) {
            int lVal = heights[l];
            int rVal = heights[r];
            int volume = min(lVal, rVal) * (r - l);
            max = volume > max ? volume : max;

            if (lVal > rVal) {
                r--;
            } else {
                l++;
            }
        }

        return max;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1

        largest_area = 0

        while r > l:
            area = min(heights[l], heights[r]) * (r - l)
            largest_area = max(area, largest_area)

            if (heights[r] < heights[l]):
                r -= 1
            else:
                l += 1
        
        return largest_area